lib/lseek64_stubs.c

```diff
@@ -46,12 +46,10 @@ CAMLprim value stub_lseek64_data(value fd, value ofs) {
   off_t c_ret;
 
   caml_enter_blocking_section();
-#if defined(SEEK_DATA)
   c_ret = lseek(c_fd, c_ofs, SEEK_DATA);
-#else
-  /* Set the file pointer to ofs; pretend there is data */
-  c_ret = lseek(c_fd, c_ofs, SEEK_SET);
-#endif
+  /* If there is no SEEK_DATA support, 
+     set the file pointer to ofs; pretend there is data */
+  if (c_ret == -1) c_ret = lseek(c_fd, c_ofs, SEEK_SET);
```

Comment:
`lseek` should return `EINVAL` if whence (in this case `SEEK_DATA`) is invalid.   Unfortunately this error code is also returned for other error conditions:
```
ERRORS
       EINVAL whence is not valid.  Or: the resulting file offset would be  negative,  or
              beyond the end of a seekable device.
```